registry.go

```diff
@@ -5,8 +5,8 @@ import "context"
 // The ServerRegistry interface is an abstraction used to expose a (potentially
 // changing) list of backend redis servers.
 type ServerRegistry interface {
-	// ListServers returns a list of redis server endpoints.
-	ListServers(ctx context.Context) ([]ServerEndpoint, error)
+	// LookupServers returns a list of redis server endpoints.
+	LookupServers(ctx context.Context) ([]ServerEndpoint, error)
```

Comment:
I renamed this from ListServers to LookupServers, I felt like it would better communicate that it may involve a network call (when using service discovery for example).